### DISCUSSION TITLE
File format for exporting certificate and end entity profiles

### DISCUSSION THREAD

**Discussion by adchst** [Keyfactor/ejbca-ce#559] [Q&A]
Hello,

when I click on export in the cert profiles or end entity profiles web page, I can download an XML file. What I'm wondering is, is this profile XML file some sort of standard or a propitiatory EJBCA format? I searched for profiles in X509 and I didn't find anything. Is this file format described in some sort of API or document?

Best regards,
Chris

**Comment by primetomas** [comment]
It's proprietary format. There is no standard for these things unfortunately. 
The XML can certainly be hard to parse.
The underlying end entity profile code is here: https://github.com/Keyfactor/ejbca-ce/blob/main/modules/ejbca-common/src/org/ejbca/core/model/ra/raadmin/EndEntityProfile.java

**Comment by svenska-primekey** [comment]
It's helpful to fill in all the parts you want in the cert profile or end entity profile. Then export and you can see how the values look and tweak accordingly to import to another instance. The Enterprise version of EJBCA has configdump which uses YAML and makes this much easier to do.
